velox/common/base/Status.h

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Adapted from Apache Arrow.
+
+#pragma once
+
+#include <fmt/format.h>
+#include <fmt/ostream.h>
```

Comment:
In fmt > 8.1 including ostream is no longer enough, you have to opt-in using a formatter specialization: https://fmt.dev/latest/api.html#ostream-api

Could you also add a specialization for the `StatusCode` enum? https://fmt.dev/latest/api.html#udt (see also #7941 for examples)

Reply:
@assignUser just added the specializations plus a few unit tests. PTAL

Reply:
Thanks, looks good, I tested the operator<< overload with fmt 10.1 and it seems it's simple enough no to need an explicit specialization: https://compiler-explorer.com/z/xGPG3oGjG :+1: